src/app/CommandSender.cpp

```diff
@@ -337,6 +340,20 @@ void CommandSender::Close()
     mTimedRequest     = false;
     MoveToState(State::AwaitingDestruction);
 
+    if (mpPendingResponseTracker)
+    {
+        Optional<uint16_t> commandRef = mpPendingResponseTracker->PopPendingResponse();
+        while(commandRef.HasValue())
+        {
+            // TODO is it okay to not provide the actual path if we have only the CommandRef?
```

Comment:
I would like opinions on this TODO. The reason why I didn't grab the `concretePath` and add it to `PendingResponseTracker` is because currently there is no plumbing to have this value in the `FinishCommand` where we add things to the tracker.

If we really do need this value to be correct when no response if give we can figure out a way, but didn't want do all the plumbing work if other felt like CommandRef is enough in this circumstance

Reply:
Not having a path in a response callback will totally break existing clients, no?  Or is the idea that they can't end up here without "opting in" somehow?

It feels like we should actually have a separate callback for the "no response" case, which would take just the CommandRef or something....  Or have the command path in the ResponseData be optional or nullable or in some other way clearly settable to "not there".

No matter what, the Callback API really needs to document what will happen here clearly.

Reply:
Went with the approach where this is a new OnNoResponse callback, that is optional for client to implement if they want to be notified about it where only CommandRef is provided